utils.ts

```diff
@@ -486,9 +489,16 @@ export async function applyPackageOverrides(
 	// pnpm@6, pnpm@7 => pnpm
 	const pm = agent?.split('@')[0]
 
-	await overridePackageManagerVersion(pkg, pm, agentVersion)
+	await overridePackageManagerVersion(pkg, pm, options.agentVersion)
 
 	if (pm === 'pnpm') {
+		const version = parseVitestVersion(options.vitestPath)
+
+		for (const key in overrides) {
+			const tar = key.replace('@', '').replace('/', '-')
+			overrides[key] = `${overrides[key]}/${tar}-${version}.tgz`
+		}
```

Comment:
How about other package managers? I'd assume they should also break when seeing `catalog:`, but it's probably not tested right now?

Reply:
Right. Other packages managers would break when seeing `catalog:`. I'd note that's not unique to the `catalog:` protocol or pnpm. The `file:`, `link:`, and `workspace:` protocols are also meant to be "local-only" and don't work across different repos. There's a requirement for a "portability" step (e.g. publish or pack) to remove protocols like `workspace:` and `catalog:`.

pnpm itself throws when encountering the `catalog:` protocol in NPM packages since we don't want to encourage non-portable specifiers being published to the NPM registry.

Reply:
I guess the test setup doesn't contain other package managers. Otherwise we should already have run into issues with `workspace:` protocol. 🤔 

I think proper solution would be to use released packages from https://pkg.pr.new - I think Vite already has that kind of workflow.